SECURITY.md

```diff
@@ -110,6 +112,42 @@ time, we will work towards improved safe-by-default configuration, but due to ba
 compatibility and performance concerns, this will require following the breaking change deprecation
 policy.
 
+#### Core and extensions
+
+Anything in the Envoy core may be used in both untrusted and trusted deployments. As a consequence,
+it should be hardened with this model in mind.  Security issues related to core code will usually
+trigger the security release process as described in this document.
+
+Some [extensions](EXTENSION_POLICY.md) are also widely used and deserve the same treatment as core.
+Examples include HTTP connection manager, TCP proxy, buffer filter, router filter, CORS, dynamic
+forward proxy, anything intended to enforce access control policies (RBAC, external authorization)
+and all transport socket extensions.
+
+Other extensions are only expected to be used in trusted environments today.  These include, but are
+not limited to most non-HTTP network filters (e.g.  Thrift, Dynamo, Mongo, Redis, Kafka, Dubbo,
+Zookeeper, MySQL), HTTP inspector, Lua and Squash. We will not treat any issues in these extensions
+as warranting invocation of the security release process.
+
+Please consult the PST if you have questions on which extensions are considered safe to use in the
+presence of untrusted traffic.
+
+#### Data and control plane
+
+We divide our threat model into data and control plane, reflecting the internal division in Envoy of
+these concepts from an architectural perspective. Our highest priority in risk assessment is the
+threat posed by untrusted downstream client traffic on the data plane. This reflects the use of
+Envoy in an edge serving capacity and also the use of Envoy as an inbound destination in a service
+mesh deployment. In addition, we also consider any vulnerability that might be exploited by
```

Comment:
re: untrusted upstreams, we already know of lots of potential issues in this area so it seems we can't claim this yet? Should we rephrase this somehow?